File: cpp/source/event/output/FacadeEvent.cpp
#include "event/output/FacadeEvent.hpp"

using namespace fm;
using namespace fm::event::output;

FacadeEvent::FacadeEvent(Type _type) :
    type(_type)
{
}

FacadeEvent::~FacadeEvent()
{
}

FacadeEvent::Type FacadeEvent::getType() const
{
    return type;
}

std::string FacadeEvent::toString() const
{
    switch (type)
    {
    case ATTACHED: return "ATTACHED";
    case RELEASED: return "RELEASED";
    case ERROR: return "ERROR";
    case MESSAGE: return "MESSAGE";
    case CHANNELS_OPENED: return "CHANNELS_OPENED";
    case OPERATION_STARTED: return "OPERATION_STARTED";
    case HANDOVER_ACCEPTED: return "HANDOVER_ACCEPTED";
    case REQUEST_CONTROL: return "REQUEST_CONTROL";
    case RELEASE_CONTROL: return "RELEASE_CONTROL";
    case HANDOVER_REJECTED: return "HANDOVER_REJECTED";
    case HANDOVER_DONE: return "HANDOVER_DONE";
    case CONTROL_RELEASED: return "CONTROL_RELEASED";
    case CHANNELS_CLOSED: return "CHANNELS_CLOSED";
    case OPERATION_ENDED: return "OPERATION_ENDED";
    case CONNECTION_LOST: return "CONNECTION_LOST";
    case CONNECTION_RECOVERED: return "CONNECTION_RECOVERED";
    case DEVICE_UNREACHABLE: return "DEVICE_UNREACHABLE";
    case LEADER_UNREACHABLE: return "LEADER_UNREACHABLE";
    case DEVICE_RECOVERED: return "DEVICE_RECOVERED";

    default:
        return "Unexpected UserEvent::Type";
    }
}

File: java/src/main/java/com/fleetmgr/sdk/client/event/output/facade/FacadeEvent.java
package com.fleetmgr.sdk.client.event.output.facade;

/**
 * Created by: Bartosz Nawrot
 * Date: 18.09.2018
 * Description:
 */
public class FacadeEvent {

    public enum Type {
        ATTACHED,
        RELEASED,
        ERROR,
        MESSAGE,
        CHANNELS_OPENED,
        OPERATION_STARTED,
        HANDOVER_ACCEPTED,
        REQUEST_CONTROL,
        RELEASE_CONTROL,
        HANDOVER_REJECTED,
        HANDOVER_DONE,
        CONTROL_RELEASED,
        CHANNELS_CLOSING,
        CHANNELS_CLOSED,
        OPERATION_ENDED,
        CONNECTION_LOST,
        CONNECTION_RECOVERED,
        DEVICE_UNREACHABLE,
        LEADER_UNREACHABLE,
        DEVICE_RECOVERED,
    }

    private Type type;

    public FacadeEvent(Type type) {
        this.type = type;
    }

    public Type getType() {
        return type;
    }

    @Override
    public String toString() {
        return "FacadeEvent{" +
                "type=" + type +
                '}';
    }
}
